### PR TITLE
ClearHistory and PageTransitions 

### LINKED ISSUE #218: Support the new router
I figured this is one of those things we haven’t gotten around to quite yet, but I thought I’d create an issue to track this.

@NathanWalker and I gave it a shot real quick the code below.

```JavaScript
import {Component} from "@angular/core";
import {HTTP_PROVIDERS} from "@angular/http";
import {Routes, ROUTER_PROVIDERS} from "@angular/router";
import {NS_ROUTER_DIRECTIVES, NS_ROUTER_PROVIDERS} from "nativescript-angular/router";
import {LoginPage} from "./pages/login/login.component";
import {ListPage} from "./pages/list/list.component";

@Component({
  selector: "main",
  directives: [NS_ROUTER_DIRECTIVES],
  providers: [HTTP_PROVIDERS, NS_ROUTER_PROVIDERS, ROUTER_PROVIDERS],
  template: "<page-router-outlet></page-router-outlet>"
})
@Routes([
  { path: "/Login", component: LoginPage },
  { path: "/List", component: ListPage }
])
export class AppComponent {}
```

But I get the error below:

```
May  6 15:52:50 107-1-111-1-ip-static sampleGroceries[6589]: CONSOLE ERROR file:///app/tns_modules/zone.js/dist/zone-node.js:421:23: Error: Uncaught (in promise): EXCEPTION: Error in :0:0
	ORIGINAL EXCEPTION: not implemented
	ERROR CONTEXT:
	[object Object]
```

I’m guessing because we don’t support the new router quite yet.

Thanks 😀

### LINKED ISSUE #284: Router with {N} parameters as clearHistory, backstackVisible
Hi, I'm trying to remove my back button after my login.

A good example is the [sample-groceries in Angular](https://github.com/NativeScript/sample-Groceries/tree/angular-end/).

After login, you can still see the back button, but I don't want that.

I am very well-known how to implement that using JS with frameModule.topmost. It's simple to add parameters such as [clearHistory](https://docs.nativescript.org/core-concepts/navigation#clear-history) or [backstackVisible](https://docs.nativescript.org/core-concepts/navigation#navigate-without-history).

Is there any way to pass these parameters with Angular?

I've tried also create a new [`<router-outlet>`](https://docs.nativescript.org/angular/core-concepts/navigation-angular#router-outlet) instead of [`<page-router-outlet>`](https://docs.nativescript.org/angular/core-concepts/navigation-angular#page-router-outlet) following the Docs about [{N} Angular Navigation](https://docs.nativescript.org/angular/core-concepts/architecture-angular.html#navigation).

So I have a login and a index for my router-outlet, however, the index should have some child routers. I'm having lot's of issues trying this solution. Actually my index doesn't replace the full page which is what I want.

Anybody with some ideas for that?

I'm happy to post some examples here...but perhaps it's a simple parameter that I couldn't find.

Also it might be related with #2170

Many thanks

### PR DESCRIPTION
**Pull request by vakrilov**
Added a way to:
 -  Clear page navigation history
 - Specify page transition on navigation.

Resolves #218
Resolves #284

### CODE REVIEW

File: nativescript-angular/router/ns-router-link.ts [hunk 1]
```diff
@@ -55,11 +64,36 @@ constructor(
...
+      transition: transition.transition
+    };
+
+    this.navigator.navigate(this.commands, extras);
+  }
 
-    this.router.navigate(
-      this.commands,
-      { relativeTo: currentRoute, queryParams: this.queryParams, fragment: this.fragment });
+  private getTrasnition(): { animated: boolean, transition?: NavigationTransition } {
```
Comment (hdeshev) [inline nativescript-angular/router/ns-router-link.ts]: typo: getTransition

**Comment by NathanWalker** [comment]
This is very exciting! Nice work @vakrilov can't wait to try this out 👍

**Comment by m-abs** [comment]
Wuhu, thank you so much. I look forward to trying this out. :+1:

**Comment by Sharique-Hasan** [comment]
Hey. I am sorry I might be doing something wrong but I could not find that. 

```
this.routerext.navigate(['/gameShowHost'], {
            transition: {
                duration: 100,
                name: 'slideLeft',
            },
            clearHistory: true
        });
```

I am working on the nativescript-ui-samples-angular app. And I can go back to the previous page.

**Comment by mhmo91** [comment]
I'm having the same issue, clear history seems to work on the native router-outlet not the page-router-outlet

**Comment by vakrilov** [comment]
Hey #mhmo91 - can you please open an issue about that (or search if there isn't already one opened).

**Comment by AkshatGiri** [comment]
I cannot find any examples on page transitions. Can anyone point me to them.

**Pull request merged**